### PR TITLE
perf(bpp): reduce computational cost

### PR DESCRIPTION
**Pull request by satoshi-ota**
## Description
Reduce aggregated costs 14.7% to **10.7%** in behavior planning container by following improvement.

- reduce frequecy to call `calcSignedArcLength`
- don't use `boost::geometry::union_`
- don't use `std::hypot`

image
image

## Tests performed

- [x] [PASS TIER IV INTERNAL SCENARIOS](https://evaluation.tier4.jp/evaluation/reports/ebe8c91f-c4ad-52b0-9b09-731b1066cecd?project_id=prd_jt) 2343/2360
- [x] [BASE LINE](https://evaluation.tier4.jp/evaluation/reports/69880819-a97b-55a2-9d09-325151af2508?project_id=prd_jt) 2332/2360

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp [hunk 1]
```diff
@@ -227,7 +227,8 @@ double calcDistanceFromPointToSegment(
 
   // target_point is outside the segment.
   return std::min(
-    tier4_autoware_utils::calcDistance2d(a, p), tier4_autoware_utils::calcDistance2d(b, p));
+    tier4_autoware_utils::calcSquaredDistance2d(a, p),
+    tier4_autoware_utils::calcSquaredDistance2d(b, p));
```
Comment (maxime-clem) [inline planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp:231]: Is it okay to return a squared distance here ?
How about putting the `std::min` inside a call to `std::sqrt` ?

Comment (danielsanchezaran) [reply planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp:231]: In calcDistanceFromPointToSegment, if the condition if (0 <= dot_val && dot_val <= squared_segment_length) is met, I believe the non squared distance is returned? 

In that case, you would be comparing squared and non-squared distances on line 242 no? 

Maybe it is better to return squared distances on both cases an rename the function to calcSquaredDistanceFromPointToSegment?

Comment (maxime-clem) [reply planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp:231]: Sure :+1:
As Daniel said, you also need to make sure the `numerator / denominator` is a squared distance for consistency.

Comment (satoshi-ota) [reply planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp:231]: @maxime-clem @danielsanchezaran Oh. I missed it. Thanks :+1:

Comment (satoshi-ota) [reply planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp]: @maxime-clem I renamed the function and fixed process consistency in https://github.com/autowarefoundation/autoware.universe/pull/6054/commits/34f0c19e6d0341a9baa5744072e9b23bc15a3ca0. Could you check that...?

Comment (maxime-clem) [reply planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp]: Looks good thank you.

**Review by maxime-clem — APPROVED** [review]

**Review by kosuke55 — APPROVED** [review]

**Review by rej55 — APPROVED** [review]

**Pull request merged**